lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosValidation.cxx

```diff
@@ -0,0 +1,184 @@
+#include "openturns/FunctionalChaosValidation.hxx"
+#include "openturns/Os.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/DesignProxy.hxx"
+#include "openturns/LeastSquaresMethod.hxx"
+#include "openturns/KFoldSplitter.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(FunctionalChaosValidation)
+
+static const Factory<FunctionalChaosValidation> Factory_FunctionalChaosValidation;
+
+/* Default constructor */
+FunctionalChaosValidation::FunctionalChaosValidation()
+  : PersistentObject()
+{}
+
+/* Default constructor */
+FunctionalChaosValidation::FunctionalChaosValidation(const FunctionalChaosResult & functionalChaosResult)
+  : PersistentObject()
+  , functionalChaosResult_(functionalChaosResult)
+{
+  // Nothing to do
+}
+
+
+/* Virtual constructor */
+FunctionalChaosValidation * FunctionalChaosValidation::clone() const
+{
+  return new FunctionalChaosValidation(*this);
+}
+
+/* String converter */
+String FunctionalChaosValidation::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << FunctionalChaosValidation::GetClassName()
+      << " functional chaos result=" << functionalChaosResult_;
+  return oss;
+}
+
+/* Get result*/
+FunctionalChaosResult FunctionalChaosValidation::getResult() const
+{
+  return functionalChaosResult_;
+}
+
+/* Compute LOO error */
+Point FunctionalChaosValidation::computeLeaveOneOutError() const
+{
+  const FunctionCollection reducedBasis(functionalChaosResult_.getReducedBasis());
+  const Function transformation(functionalChaosResult_.getTransformation());
+  const Sample inputSample(functionalChaosResult_.getInputSample());
+  const Sample standardSample(transformation(inputSample));
+  DesignProxy designProxy(standardSample, reducedBasis);
+  const UnsignedInteger reducedBasisSize = reducedBasis.getSize();
+  Indices allIndices(reducedBasisSize);
+  allIndices.fill();
+  const MatrixImplementation designMatrix(designProxy.computeDesign(allIndices));
+  // The method name is set to the default one, given by ResourceMap
+  String methodName = ResourceMap::GetAsString("LeastSquaresExpansion-DecompositionMethod");
+  LeastSquaresMethod leastSquaresMethod = LeastSquaresMethod::Build(methodName, designProxy, allIndices);
+  const Sample outputSample(functionalChaosResult_.getOutputSample());
+  const Function metamodel(functionalChaosResult_.getMetaModel());
+  const Sample outputMetamodel(metamodel(inputSample));
+  const UnsignedInteger outputDimension = outputSample.getDimension();
+  const UnsignedInteger sampleSize = inputSample.getSize();
+  Point leaveOneOutMSE(outputDimension);
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+  {
+    const Sample marginalOutputSample(outputSample.getMarginal(j));
+    const Point marginalOutputPoint(marginalOutputSample.asPoint());
+    // Necessary to compute the projection matrix
+    const Point betaHat = leastSquaresMethod.solve(marginalOutputPoint);
+    const Point diagonalH = leastSquaresMethod.getHDiag();
```

Comment:
This is costly and should be done outside a loop as it is independent from output marginal!

Reply:
Fixed at 2df23ef8e1fd48325d0a47642892d953c24e9590